### PR TITLE
feat: implement teeps feedback

### PR DESCRIPTION
**Pull request by ryanwolhuter**
address feedback from teep described here: https://www.notion.so/makerdao-ses/07b13109dc794e23aa01016dab616a19?v=30898d62283f4309ae76f0b8d96c0086&p=13643679273847d8a9407cfde26efea2&pm=s

Some of these updates need to be done in document-model-libs. Here is a list of the changes done in this PR:

- The ***quantity*** and ***asset proceeds*** fields don’t accept comma’s
- The ***cash balance change*** should be formatted like a currency value (commas)
- Same for the number values in the txn table - they should have ,000 separators
- Same for number values in the portfolio table
- Consistency for the decimal points (either all to 2 (0.00) or none) - preference to have all to 0.00
- No realized surplus in the portfolio table - was this a design decision?

### CODE REVIEW

File: src/rwa/components/inputs/number-input.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import { NumericFormat } from 'react-number-format';
+import { RWATableTextInput } from '.';
+
+type Props<ControlInputs extends FieldValues> = ComponentPropsWithRef<
+    typeof RWATableTextInput
+> & {
+    name: Path<ControlInputs>;
+    control: Control<ControlInputs>;
+    disabled?: boolean;
+    requiredErrorMessage?: string;
```
Comment (ryanwolhuter) [inline src/rwa/components/inputs/number-input.tsx:12]: Passing a string for an error message in to the `required` rule is a shortcut for setting the rule to true and providing the message

File: src/rwa/components/inputs/number-input.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        control,
+        requiredErrorMessage,
+        currency,
+        value: _,
+        onChange: __,
+        numericFormatProps,
+        ...inputProps
+    } = props;
+
+    const {
```
Comment (ryanwolhuter) [inline src/rwa/components/inputs/number-input.tsx:31]: set sensible defaults as described by teep

File: src/rwa/components/inputs/number-input.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import { RWATableTextInput } from '.';
+
+type Props<ControlInputs extends FieldValues> = ComponentPropsWithRef<
+    typeof RWATableTextInput
+> & {
+    name: Path<ControlInputs>;
+    control: Control<ControlInputs>;
+    disabled?: boolean;
+    requiredErrorMessage?: string;
+    currency?: 'USD';
```
Comment (ryanwolhuter) [inline src/rwa/components/inputs/number-input.tsx:13]: for now 'USD' is the only one supported, but we can expand this list when needed

File: src/rwa/components/table/formatted-number.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import { NumericFormat } from 'react-number-format';
+
+type Props = Omit<
+    ComponentPropsWithoutRef<typeof NumericFormat>,
+    'displayType'
+> & {
+    currency?: 'USD';
+};
+
+export function FormattedNumber(props: Props) {
```
Comment (ryanwolhuter) [inline src/rwa/components/table/formatted-number.tsx:11]: wrapper to make it easy to use the default props as described by teep. `displayType="text"` just renders a span.

**Review by acaldas — APPROVED** [review]

**Review by gpuente — APPROVED** [review]

**Pull request merged**